flexget/plugins/filter/movie_queue.py

```diff
@@ -396,3 +428,277 @@ def queue_get(session=None, downloaded=False):
+    ]
+}
+
+movie_edit_results_schema = api.schema('movie_edit_results_schema', movie_edit_results_schema)
+movie_edit_input_schema = api.schema('movie_edit_input_schema', movie_edit_input_schema)
+
+
+@movie_queue_api.route('/')
+class MovieQueueAPI(APIResource):
+    @api.response(404, 'Page does not exist')
+    @api.response(200, 'Movie queue retrieved successfully', movie_queue_schema)
+    @api.doc(parser=movie_queue_parser)
+    def get(self, session=None):
+        """ List queued movies """
+        args = movie_queue_parser.parse_args()
+        page = args['page']
+        max_results = args['max']
+        downloaded = args['downloaded']
+
+        movie_queue = queue_get(session=session, downloaded=downloaded)
+        count = len(movie_queue)
+
+        pages = int(ceil(count / float(max_results)))
+
+        movie_items = []
+
+        if page > pages and pages != 0:
+            return {'error': 'page %s does not exist' % page}, 404
+
+        start = (page - 1) * max_results
+        finish = start + max_results
+        if finish > count:
+            finish = count
+
+        for movie_number in range(start, finish):
+            movie_items.append(movie_queue[movie_number])
+
+        return jsonify({
+            'movies': [movie.to_dict() for movie in movie_items],
+            'number_of_movies': count,
+            'page_number': page,
+            'total_number_of_pages': pages
+        })
+
+    @api.response(400, 'Page not found')
+    @api.response(200, 'Movie successfully added', movie_add_results_schema)
+    @api.validate(movie_add_input_schema)
+    def post(self, session=None):
+        """ Add movies to movie queue """
+        kwargs = request.json
+
+        try:
+            kwargs['quality'] = qualities.Requirements(kwargs.get('quality'))
+        except ValueError as e:
+            reply = {
+                'status': 'error',
+                'message': e.message
+            }
+            return reply, 400
+        kwargs['session'] = session
+
+        try:
+            movie = queue_add(**kwargs)
+        except QueueError as e:
+            reply = {
+                'status': 'error',
+                'message': e.message
+            }
+            return reply, 400
+
+        return jsonify(
+            {
+                'message': 'Successfully added movie to movie queue',
+                'movie': movie
+            }
+        )
+
+    @api.response(400, 'Page not found')
+    @api.response(200, 'Movie successfully deleted', movie_del_results_schema)
+    @api.validate(movie_del_input_schema)
+    def delete(self, session=None):
+        """ Delete movies from movie queue """
+        kwargs = request.json
+
+        kwargs['session'] = session
+
+        try:
+            movie = queue_del(**kwargs)
+        except QueueError as e:
+            reply = {
+                'status': 'error',
+                'message': e.message
+            }
+            return reply, 400
```

Comment:
What could be the possible errors here?? Maybe we need to return a 500 instead of a 400??

400 = client errors
500 = server errors